emcc.py

```diff
@@ -2555,11 +2555,15 @@ def phase_linker_setup(options, state, newargs):
   if settings.AUDIO_WORKLET:
     if settings.AUDIO_WORKLET == 1:
       settings.AUDIO_WORKLET_FILE = unsuffixed(os.path.basename(target)) + '.aw.js'
+      if not settings.MINIMAL_RUNTIME:
+        # MINIMAL_RUNTIME exports these manually, since this export mechanism is placed
+        # in global scope that is not suitable for MINIMAL_RUNTIME loader.
+        settings.EXPORTED_RUNTIME_METHODS += ['stackSave', 'stackAlloc', 'stackRestore']
+    elif settings.AUDIO_WORKLET == 2:
+      settings.BINARYEN_ASYNC_COMPILATION = 0 # Run synchronous Wasm initialization inside AudioWorklet.
+      settings.SINGLE_FILE = 1 # All code must be embedded in a single .js file when targeting -sAUDIO_WORKLET=2
```

Comment:
I don't think this needs to be a requirement. I think a better pattern is to fetch the wasm binary on the main thread and then transfer it to the AudioWorklet processor.